### PR TITLE
ARROW-15122: [R] Gate parquet tests on snappy

### PR DESCRIPTION
**Pull request by jonkeane**
We could also remove [the patch step entirely](https://github.com/apache/arrow/blob/f473cb908f4b25e3d5e4343af5c4b567751c0995/cpp/cmake_modules/ThirdpartyToolchain.cmake#L624-L630) if we don't want to even try to avoid the UBSan.

### CODE REVIEW

File: r/R/arrow-package.R [hunk 1]
```diff
@@ -153,6 +153,9 @@ arrow_with_dataset <- function() {
 #' @rdname arrow_available
 #' @export
 arrow_with_parquet <- function() {
+  # https://github.com/google/snappy/pull/148/files
+  skip_on_linux_devel()
```
Comment (nealrichardson) [inline r/R/arrow-package.R:157]: This doesn't work: that function is in a test helper, and it requires testthat

Comment (nealrichardson) [reply r/R/arrow-package.R:157]: Also I don't think we have to gate all of parquet support on this, only snappy

Comment (jonkeane) [reply r/R/arrow-package.R:157]: Right, right. I'll pull it out into a separate helper (the renaming of the skip makes it more accurate with what it does though so I'm inclined to keep that)

And, yeah we don't need to gate all of parquet, though anywhere we don't set a default will use snappy. I'm not 100% sure why [only the example](https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=16490&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181&l=12300) runs into the UBSan, possibly the shape of data / how we are projecting is unique there?

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer test-fedora-r-clang-sanitizer test-r-linux-valgrind test-r-linux-as-cran

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer

Try once more to see if we can overcome the docker image timeout issue

**Comment by jonkeane** [comment]
Ok, this is ready for review. The test-r-linux-as-cran test failures are before the tests even run (and are on master too...)

**Comment by jonkeane** [comment]
@github-actions crossbow submit -g r

**Comment by jonkeane** [comment]
@crossbow submit test-r-linux-valgrind test-r-minimal-build test-r-offline-minimal test-r-rhub-ubuntu-gcc-release-latest test-r-rstudio-r-base-4.1-centos7-devtoolset-8 test-r-rstudio-r-base-4.1-centos8 test-r-rstudio-r-base-4.1-focal test-r-rstudio-r-base-4.1-opensuse42 test-ubuntu-18.04-r-sanitizer test-r-versions 

Re-running these to see if we can overcome packagemanager.rstudio.com from timing out / failing

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-r-linux-valgrind test-r-minimal-build test-r-offline-minimal test-r-rhub-ubuntu-gcc-release-latest test-r-rstudio-r-base-4.1-centos7-devtoolset-8 test-r-rstudio-r-base-4.1-centos8 test-r-rstudio-r-base-4.1-focal test-r-rstudio-r-base-4.1-opensuse42 test-ubuntu-18.04-r-sanitizer test-r-versions

Re-running these to see if we can overcome packagemanager.rstudio.com from timing out / failing with the actual command 🤦

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-r-linux-valgrind test-r-rhub-ubuntu-gcc-release-latest test-r-rstudio-r-base-4.1-centos7-devtoolset-8 test-r-rstudio-r-base-4.1-centos8 test-r-rstudio-r-base-4.1-focal test-r-rstudio-r-base-4.1-opensuse42 test-ubuntu-18.04-r-sanitizer test-r-versions

RSPM looks like it's back up? Also the minimal builds are being fixed elsewhere

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer test-r-linux-valgrind test-fedora-r-clang-sanitizer

Double checking without installing patch that these are ok

**Comment by nealrichardson** [comment]
I'd remove the patch step entirely. May also want to make snappy tests skip on CRAN just like we do re2.

**Comment by wjones127** [comment]
Is this a temporary measure? Do we want to keep the test "Error messages are shown when the compression algorithm snappy is not found" in test-parquet.R?

**Comment by jonkeane** [comment]
> Is this a temporary measure?

Hopefully it is — we have a PR upstream with snappy to resolve the UBSan issue, but that hasn't been merged/released yet

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer test-r-linux-valgrind test-fedora-r-clang-sanitizer test-r-linux-as-cran

and without the patch at all

**Pull request closed without merging**